src/class_sqlThread.py

```diff
@@ -17,128 +19,104 @@
-            self.conn.commit()
-            logger.info('Created messages database file')
+            with open(os.path.join(self.root_path, "pybitmessage/sql/{}.sql".format(file_name))) as sql_file:
+                sql_as_string = sql_file.read()
+            self.cur.executescript(sql_as_string)
+        except IOError as err:
+            logger.debug(
+                'ERROR trying to initialize database. Error message: %s\n', str(err))
         except Exception as err:
-            if str(err) == 'table inbox already exists':
-                logger.debug('Database file already exists.')
-
-            else:
-                sys.stderr.write(
-                    'ERROR trying to create database file (message.dat). Error message: %s\n' % str(err))
-                os._exit(0)
-
-        # If the settings version is equal to 2 or 3 then the
-        # sqlThread will modify the pubkeys table and change
-        # the settings version to 4.
-        settingsversion = config.getint(
-            'bitmessagesettings', 'settingsversion')
-
-        # People running earlier versions of PyBitmessage do not have the
-        # usedpersonally field in their pubkeys table. Let's add it.
-        if settingsversion == 2:
-            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 3
-
-        # People running earlier versions of PyBitmessage do not have the
-        # encodingtype field in their inbox and sent tables or the read field
-        # in the inbox table. Let's add them.
-        if settingsversion == 3:
-            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 4
-
-        config.set(
-            'bitmessagesettings', 'settingsversion', str(settingsversion))
-        config.save()
-
-        helper_startup.updateConfig()
-
-        # From now on, let us keep a 'version' embedded in the messages.dat
-        # file so that when we make changes to the database, the database
-        # version we are on can stay embedded in the messages.dat file. Let us
-        # check to see if the settings table exists yet.
-        item = '''SELECT name FROM sqlite_master WHERE type='table' AND name='settings';'''
-        parameters = ''
-        self.cur.execute(item, parameters)
+            logger.debug(
+                'ERROR trying to initialize database. Error message: %s\n', str(err))
+
```

Comment:
We should also handle `FileNotFoundError`, then log debug that the file is missing, and return.

Reply:
I have checked that `FileNotFoundError` is for `python3`.
or we can do like
`if six.PY2: 
    FileNotFoundError = IOError`

or
`try:
    FileNotFoundError
except NameError:
    # py2
    FileNotFoundError = IOError`

And `IOError` raises `IOError(2, 'No such file or directory')`